docs/HailExpressionLanguage.md

```diff
@@ -56,6 +56,8 @@ Several Hail commands provide the ability to perform a broad array of computatio
      - max: `arr.max` -- valid only for numeric arrays, returns the minimum value
      - find: `arr.find(v => expr)` -- Returns the first non-missing element of `arr` for which `expr` is true.  If no element satisfies the predicate, `find` returns NA.
      - map: `arr.map(v => expr)` -- Returns a new array produced by applying `expr` to each element
+     - flatMap: `arr.flatMap(v => expr)` -- valid only for `expr` of type Array. Returns a new array by mapping each element of `arr` and concatenating the resulting arrays
```

Comment:
It's worth thinking about the case where the elements in the sets aren't unique

```
scala> Set(Array(1, 2), Array(2, 3)).flatten
res0: scala.collection.immutable.Set[Int] = Set(1, 2, 3)

scala> Array(Set(1, 2), Set(2, 3)).flatten
res1: Array[Int] = Array(1, 2, 2, 3)

scala> Set(Array(1, 2), Array(2, 3)).flatMap(x => x)
res2: scala.collection.immutable.Set[Int] = Set(1, 2, 3)

scala> Array(Set(1, 2), Set(2, 3)).flatMap(x => x)
res3: Array[Int] = Array(1, 2, 2, 3)
```

It seems that Scala defines flatMap something like:

```scala
trait Monad[T] {
  def flatMap[B[_], U](f : T => B[U])(implicit convert: ConvertableTo[B[U], Monad[U]]));
}
```

This seems to be the age-old question of implicit conversions. I suppose in an "interactive" language, you'd want sensible defaults rather than safety belts.